src/react/DateTime/index.tsx

```diff
@@ -244,11 +241,9 @@ const DateTime = ({
                   if (
                     relatedTarget !== calendar && !calendar.contains(relatedTarget as HTMLElement)
                   ) {
-                    if (returnDate) {
-                      onChange && onChange(returnDate);
-                      setShowPickerError(false);
-                    }
                     setPickerDisplay('none');
+                    onChange && onChange(returnDate);
+                    setShowPickerError(false);
```

Comment:
Why do you want remove if condition ?
Please keep old logic if it unnecessary to remove